lib/ecto_diff.ex

```diff
@@ -45,13 +45,52 @@ defmodule EctoDiff do
   """
   @type t :: %__MODULE__{
           struct: atom(),
-          primary_key: %{required(atom()) => any()},
-          changes: %{required(atom()) => any()},
+          primary_key: %{required(struct_field) => any()},
+          changes: %{required(struct_field) => any()},
           effect: effect(),
           previous: Ecto.Schema.t(),
           current: Ecto.Schema.t()
         }
 
+  @typedoc """
+  Configurable options for `diff/3`.
+
+  ## Options
+
+  * `:overrides` - A keyword list or map which provides a reference from a struct
+    to a key (or list of keys) on that struct which will be used as the primary key
+    (simple or composite) for diffing.
+  """
+  @type diff_opts :: [
+          overrides: overrides
+        ]
+
+  @typedoc """
+  A keyword list or a map which specifies an override from an Ecto schema to the desired
+  primary key, for use in comparing structs.
+
+  Structs that are not specified will be compared using their default primary key.
+
+  ## Examples:
+
+      [{Pet, :refid}, {Skill, :id}]
+
+  or
+
+      %{Skill => :refid, Owner => [:id, :refid]}
+  """
+  @type overrides :: [{Ecto.Schema.t(), primary_key}] | %{Ecto.Schema.t() => primary_key}
```

Comment:
This type needs to be updated to an atom.
```suggestion
  @type overrides :: [{atom(), primary_key}] | %{atom() => primary_key}
```